segmentstore/server/src/test/java/io/pravega/segmentstore/server/DataRecoveryTestUtils.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+import static io.pravega.shared.NameUtils.getMetadataSegmentName;
+
+/**
+ * Utility methods for data recovery tests.
+ */
+@Slf4j
+public class DataRecoveryTestUtils {
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+    private static final ScheduledExecutorService EXECUTOR_SERVICE = createExecutorService(10);
+
+    /**
+     * Lists all segments from a given long term storage.
+     * @param tier2             Long term storage.
+     * @param containerCount    Total number of segment containers.
+     * @return                  A map of lists containing segments by container Ids.
+     * @throws                  IOException in case of exception during the execution.
+     */
+    public static Map<Integer, List<SegmentProperties>> listAllSegments(Storage tier2, int containerCount) throws IOException {
+        SegmentToContainerMapper segToConMapper = new SegmentToContainerMapper(containerCount);
+        Map<Integer, List<SegmentProperties>> segmentToContainers = new HashMap<Integer, List<SegmentProperties>>();
+        log.info("Generating container files with the segments they own...");
+        Iterator<SegmentProperties> it = tier2.listSegments();
+        if (it == null) {
+            return segmentToContainers;
+        }
+        // Iterate through all segments. Put each one of them in its respective list.
+        while (it.hasNext()) {
+            SegmentProperties curr = it.next();
+            int containerId = segToConMapper.getContainerId(curr.getName());
+            List<SegmentProperties> segmentsList = segmentToContainers.get(containerId);
+            if (segmentsList == null) {
+                segmentsList = new ArrayList<>();
+                segmentsList.add(curr);
+                segmentToContainers.put(containerId, segmentsList);
+            } else {
+                segmentToContainers.get(containerId).add(curr);
+            }
+        }
+        return segmentToContainers;
+    }
+
+    public static ScheduledExecutorService createExecutorService(int threadPoolSize) {
+        ScheduledThreadPoolExecutor es = new ScheduledThreadPoolExecutor(threadPoolSize);
+        es.setContinueExistingPeriodicTasksAfterShutdownPolicy(false);
+        es.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
+        es.setRemoveOnCancelPolicy(true);
+        return es;
+    }
+
+     /**
+     * Creates all segments given in the list with the given DebugStreamSegmentContainer.
+     */
+     public static class Worker implements Runnable {
+        private final int containerId;
+        private final DebugStreamSegmentContainer container;
+        private final List<SegmentProperties> segments;
+        public Worker(DebugStreamSegmentContainer container, List<SegmentProperties> segments) {
```

Comment:
What is this worker and why is it in your `test` package? Isn't this doing the bulk of your recovery work?

Reply:
It is only used by tests in integration and StreamSegmentTestBase. Though, I changed it's name.

Reply:
Should I move it to main? I think I will need to reuse it in CLI tool.